### PR TITLE
feature/TINY-7052: BDD conversion: Insert Date Time

### PR DESCRIPTION
**Pull request by camiloHimura**
Related Ticket: [7052](https://ephocks.atlassian.net/secure/RapidBoard.jspa?rapidView=116&modal=detail&selectedIssue=TINY-7052)

Description of Changes:
* Refactor plugins to BDD-style tests

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] License headers added on new files (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts [hunk 1]
```diff
@@ -1,43 +1,38 @@
...
+import Editor from 'tinymce/core/api/Editor';
 
-import InsertDatetimePlugin from 'tinymce/plugins/insertdatetime/Plugin';
-import SilverTheme from 'tinymce/themes/silver/Theme';
+import Plugin from 'tinymce/plugins/insertdatetime/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
 
-UnitTest.asynctest('browser.tinymce.plugins.insertdatetime.InsertDatetimeSanityTest', (success, failure) => {
+describe('browser.tinymce.plugins.insertdatetime.InsertDatetimeSanityTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
```
Comment (HAFRMO) [inline modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts:10]: Can this be moved to a `before` or `beforeEach`?

Comment (JamesToohey) [reply modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts:10]: The hook needs to sit inside the `describe` like this to work

File: modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts [hunk 2]
```diff
@@ -1,43 +1,38 @@
...
-  }, success, failure);
-}
-);
+          })
+        ]
+      });
+    }));
+  });
+});
+
```
Comment (JamesToohey) [inline modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts:38]: I think this is an extra newline (not including the one at the EOF)

**Review by JamesToohey — APPROVED** [review]

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts:10]: Also worth noting that `TinyHooks` is an implementation that uses the before, etc... hooks, so it's already using that and this is how you access the data setup in those hooks without having to redeclare the same TinyMCE setup logic everytime (ie reduces duplication).

File: modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts [hunk 3]
```diff
@@ -1,43 +1,38 @@
...
-import { TinyApis, TinyLoader, TinyUi } from '@ephox/mcagar';
+import { ApproxStructure } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyHooks, TinyUiActions } from '@ephox/mcagar';
+import Editor from 'tinymce/core/api/Editor';
 
-import InsertDatetimePlugin from 'tinymce/plugins/insertdatetime/Plugin';
-import SilverTheme from 'tinymce/themes/silver/Theme';
+import Plugin from 'tinymce/plugins/insertdatetime/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
```
Comment (lnewson) [inline modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts:7]: The `Editor` import is in the wrong location here, as it should be alongside the `Plugin`/`Theme` imports. eg:
```
import { TinyAssertions, TinyHooks, TinyUiActions } from '@ephox/mcagar';

import Editor from 'tinymce/core/api/Editor';
import Plugin from 'tinymce/plugins/insertdatetime/Plugin';
import Theme from 'tinymce/themes/silver/Theme';
```

- [ ] Fix imports

File: modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts:38]: When you open this directly it looks fine, so I think it's GitHub just being weird... https://github.com/tinymce/tinymce/blob/fdd93a23d03325bae82b57e84dba33d7a93abf11/modules/tinymce/src/plugins/insertdatetime/test/ts/browser/InsertDatetimeSanityTest.ts

**Review by lnewson — APPROVED** [review]

**Pull request merged**